resource_management_mobile/src/app/pages/location/location.page.ts

```diff
@@ -1,15 +1,244 @@
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { ExportOptionComponent } from 'src/app/shared/components/export-option/export-option.component';
 
 @Component({
   selector: 'app-location',
   templateUrl: './location.page.html',
   styleUrls: ['./location.page.scss'],
 })
 export class LocationPage implements OnInit {
-
-  constructor() { }
+  locationForm!: FormGroup;
+  showSearch: boolean = false;
+  locationData: any[] = [];
+  skip: number = 0;
+  searchQuery: string = '';
+  isModalOpen: boolean = false;
+  modelType!: string;
+  locationMoreData!: any;
+  locationEdit: boolean = false;
+  isEnableEdit: boolean = false;
+  selectedIndex!: number;
+  constructor(
+    private locationService: LocationService,
+    private toastService: ToastService,
+    private modalCtrl: ModalController,
+    private toastConstants: ToastConstants
+  ) { }
 
   ngOnInit() {
+    this.getLocation(this.skip, 20, this.searchQuery);
+    this.locationForm = new FormGroup({
+      Description: new FormControl('', Validators.required),
+    });
+  }
+
+  onSearch() {
+    this.showSearch = !this.showSearch;
+  }
+
+  getLocation(skip: number, limit: number, search: string) {
+    this.locationService
+      .getLocation(skip, limit, search)
+      .subscribe((data: locationResponce) => {
+        this.locationData = [...this.locationData, ...data.data.locationInfo];
+      });
+  }
+
+  onIonInfinite(ev: any) {
+    this.skip = this.skip + 20;
+    this.getLocation(this.skip, 20, this.searchQuery);
+    setTimeout(() => {
+      (ev as InfiniteScrollCustomEvent).target.complete();
+    }, 500);
+  }
+
+  setOpen(isOpen: boolean, type: string, locationInfo?: any, index?: any) {
+    this.modelType = type;
+    this.isModalOpen = isOpen;
+    this.locationEdit = false;
+    this.locationMoreData = locationInfo;
+    this.selectedIndex = index;
+    this.setDataLocationForm(locationInfo);
+  }
+
+  setDataLocationForm(locationInfo: any) {
+    if (locationInfo != undefined) {
+      this.isEnableEdit = true;
+      this.locationForm.patchValue({
+        Description: locationInfo.Description,
+      });
+    } else {
+      this.isEnableEdit = false;
+      this.locationEdit = false;
+      this.locationForm.reset();
+    }
+  }
+
+  handleSearch(event: any) {
+    this.searchQuery = event.target.value.toLowerCase();
+    this.locationData = [];
+    this.skip = 0;
+    this.getLocation(this.skip, 20, this.searchQuery);
+  }
+
+  deleteLocation(id: string, index: number) {
+    this.locationService.deleteLocation(id).subscribe({
+      next: (response) => {
+        response = response as locationResponce;
+        this.locationData.splice(index, 1);
+        this.toastService.presentToast(
+          this.toastConstants.Delete_success_message
+        );
+      },
+      error: (response) => {
+        this.toastService.presentToast(this.toastConstants.try_again);
```

Comment:
Instead of "presentToast" use "errorToast" from toast service.

Reply:
changed